gpudrive/integrations/il/storage.py

```diff
@@ -0,0 +1,222 @@
+            veh_collision = infos.collided[cont_agent_mask]
+            goal_achieved = infos.goal_achieved[cont_agent_mask]
+
+            off_road_rate = off_road.sum().float() / cont_agent_mask.sum().float()
+            veh_coll_rate = veh_collision.sum().float() / cont_agent_mask.sum().float()
+            goal_rate = goal_achieved.sum().float() / cont_agent_mask.sum().float()
+            collision_rate = off_road_rate + veh_coll_rate
+            collision = (veh_collision + off_road > 0)
+            print(f'Offroad {off_road_rate} VehCol {veh_coll_rate} Goal {goal_rate}')
+            print(f'Save number w/o collision {len(expert_global_pos_lst[~collision])} / {len(expert_global_pos_lst)}')
+            break
+
+    expert_global_pos_lst = expert_global_pos_lst[~collision].to('cpu')
+    expert_global_rot_lst = expert_global_rot_lst[~collision].to('cpu')
+    
+    os.makedirs(save_path, exist_ok=True)
+    np.savez_compressed(f"{save_path}/global_trajectory_{save_index}.npz", 
+                        ego_global_pos=expert_global_pos_lst,
+                        ego_global_rot=expert_global_rot_lst)
+
+
+if __name__ == "__main__":
+    import argparse
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--num_stack', type=int, default=1)
+    parser.add_argument('--save_path', type=str, default='/')
+    parser.add_argument('--save_index', type=int, default=0)
+    parser.add_argument('--dataset', type=str, default='train', choices=['train', 'valid'],)
+    parser.add_argument('--function', type=str, default='save_global_pos_and_rot', 
+                        choices=[
+                            'save_trajectory_and_three_mask_by_scenes',
+                            'save_global_pos_and_rot'])
+    parser.add_argument('--dataset-size', type=int, default=1000) # total_world
+    parser.add_argument('--batch-size', type=int, default=5) # num_world
+    args = parser.parse_args()
+
+    torch.set_printoptions(precision=3, sci_mode=False)
+    print()
+    print("num_stack : ", args.num_stack)
+    print("save_path : ", args.save_path)
+    print("save_index : ", args.save_index)
+    print("dataset : ", args.dataset)
+    print("function : ", args.function)
+    # Initialize configurations
+    env_config = EnvConfig(
+        dynamics_model='delta_local',
+        steer_actions=torch.round(torch.tensor([-np.inf, np.inf]), decimals=3),
+        accel_actions=torch.round(torch.tensor([-np.inf, np.inf]), decimals=3),
+        dx=torch.round(torch.tensor([-6.0, 6.0]), decimals=3),
+        dy=torch.round(torch.tensor([-6.0, 6.0]), decimals=3),
+        dyaw=torch.round(torch.tensor([-np.pi, np.pi]), decimals=3),
+    )
+    print('Scene Loader')
+    # Create data loader
+    train_loader = SceneDataLoader(
+        root="data/processed/examples", #f"/data/formatted_json_v2_no_tl_{args.dataset}/",
+        batch_size=args.batch_size,
+        dataset_size=args.dataset_size,
+        sample_with_replacement=True,
```

Comment:
True이면 랜덤하게 Scene 가져오는거 아닌가요?
```suggestion
        sample_with_replacement=False,
```

Reply:
다시 올린 거에서 고침